libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/data/ViewportDataSourceProcessor.kt

```diff
@@ -239,18 +240,25 @@ internal object ViewportDataSourceProcessor {
      * Returns the pitch based on route progress and settings.
      */
     fun getPitchFallbackFromRouteProgress(
-        pitchNearManeuversEnabled: Boolean,
-        triggerDistanceForPitchZero: Double,
-        defaultPitch: Double,
-        distanceRemainingOnStep: Float
+        routeProgress: RouteProgress,
+        followingFrameOptions: FollowingFrameOptions
     ): Double {
-        return if (
-            pitchNearManeuversEnabled &&
-            distanceRemainingOnStep <= triggerDistanceForPitchZero
-        ) {
-            MapboxNavigationViewportDataSource.ZERO_PITCH
-        } else {
-            defaultPitch
+        val currentStepProgress = routeProgress.currentLegProgress?.currentStepProgress
+        val upcomingManeuverType = routeProgress.bannerInstructions?.primary()?.type()
```

Comment:
Is there an easier way to find the upcoming maneuver?

Reply:
I don't think so.